src/plugins/provenance_rename_operator.py

```diff
@@ -14,33 +14,80 @@ class ProvenanceRenameOperator(BaseOperator):
     @apply_defaults
     def __init__(
         self,
-        dataset_name,
-        pg_schema="public",
-        postgres_conn_id="postgres_default",
-        rename_indexes=False,
-        prefix_table_name="",
-        postfix_table_name="",
-        *args,
-        **kwargs,
-    ):
+        dataset_name: str,
+        pg_schema: str = "public",
+        postgres_conn_id: str = "postgres_default",
+        rename_indexes: bool = False,
+        prefix_table_name: str = "",
+        postfix_table_name: str = "",
+        table_name: str = "",
+        *args: Any,
+        **kwargs: Dict,
+    ) -> None:
         super().__init__(*args, **kwargs)
-        self.postgres_conn_id = postgres_conn_id
-        self.dataset_name = dataset_name
-        self.pg_schema = pg_schema
-        self.rename_indexes = rename_indexes
-        # The table to enforce the provenance translations is defined by the table ID in the schema definition.
-        # If the provenance translations must be applied on a temp table name i.e. 'spoorlijnen_metro_new'
-        # then specify the prefix (i.e. spoorlijnen_) and postfix (i.e. _new) when calling this operator.
-        self.prefix_table_name = prefix_table_name
-        self.postfix_table_name = postfix_table_name
-
-    def _snake_tablenames(self, tablenames):
+        self.postgres_conn_id: str = postgres_conn_id
+        self.dataset_name: str = dataset_name
+        self.pg_schema: str = pg_schema
+        self.rename_indexes: bool = rename_indexes
+        # The table to enforce the provenance translations is defined by
+        # the table ID in the schema definition. If the provenance translations
+        # must be applied on a temp table name i.e. 'spoorlijnen_metro_new'. Then specify
+        # the prefix (i.e. spoorlijnen_) and postfix (i.e. _new) when calling this operator.
+        self.prefix_table_name: str = prefix_table_name
+        self.postfix_table_name: str = postfix_table_name
+        # set table name for getting provenance for specific table
+        self.table_name: str = table_name
+
+    def _snake_tablenames(self, tablenames: Iterable[Union[List[Any], KeysView[Any]]]) -> str:
```

Comment:
Wellicht zou `KeysView` met een gewone `list` kunnen. De `KeysView` is vanaf python 3.9 deprecated.

Reply:
Zowel `List` als `KeysView` hebben in hun type hierarchie `Iterable` zitten. Dus de `Union` kan je versimpelen tot: `Iterable`. Dus dan zou je krijgen:

```python
tablenames: Iterable[Iterable[Any]]
```
maar dan lijkt me niet correct. Is het niet gewoon:
```python
tablenames: Iterable[str]
```
???

Reply:
BTW als je `Any` kan vermijden is dat iets goed. Met `Any` gooi je bijna alle type checking de deur uit.

Reply:
Tav hierarchie. Zie ook de tabel op https://docs.python.org/3.8/library/collections.abc.html

Reply:
:+1:  aangepast en goed te weten dat Any zoveel mogelijk vermijd moet worden